scripts/qc.R

```diff
@@ -0,0 +1,306 @@
+              help="alt column [default=%default]", metavar="character"),
+  make_option(c("--af_alt_col_suffix"), type="character", default="_af_alt",
+              help="af alt column suffix [default=%default]", metavar="character"),
+  make_option(c("--pheno"), type="character", default="pheno",
+              help="phenotype name [default=%default]", metavar="character")
+);
+
+opt_parser <- OptionParser(option_list = option_list)
+opt <- parse_args(opt_parser, positional_arguments = 0)
+
+options(bitmapType = "cairo")
+
+print(str(opt))
+chr_col <- opt$options$chr_col
+bp_col <- opt$options$bp_col
+ref_col <- opt$options$ref_col
+alt_col <- opt$options$alt_col
+af_alt_col_suffix <- opt$options$af_alt_col_suffix
+pheno <- opt$options$pheno
+method <- opt$options$method
+pval_thresh <- sort(unique(as.numeric(unlist(strsplit(gsub("[[:blank:]]", "", opt$options$pval_thresh), ",")))), decreasing = T)
+leave <- opt$options$loo
+region <- round(opt$options$region * 10^6 / 2, 0)
+
+file <- opt$options$file
+conf <- rjson::fromJSON(file = opt$options$conf)
+
+qc_dt <- data.table(pheno = pheno)
+for (s in conf$meta) {
+  qc_dt[, (paste0(s$name, "_n_cases")) := s$n_cases]
+  qc_dt[, (paste0(s$name, "_n_controls")) := s$n_controls]
+}
+
+studies <- sapply(conf$meta, function(x) x$name)
+
+study_pval_cols <- sapply(conf$meta, function(x) paste(x$name, x$pval, sep = "_"))
+meta_pval_col <- paste("all", method, "meta_p", sep = "_")
+
+study_beta_cols <- sapply(conf$meta, function(x) paste(x$name, x$effect, sep = "_"))
+meta_beta_col <- paste("all", method, "meta_beta", sep = "_")
+
+het_p_col <- paste("all", method, "het_p", sep = "_")
+
+af_cols <- paste0(studies, af_alt_col_suffix)
+
+pval_cols <- c(study_pval_cols, meta_pval_col)
+meta_pval_cols <- meta_pval_col
+
+keep_cols <- c(chr_col, bp_col, ref_col, alt_col, af_cols, study_pval_cols, meta_pval_col, study_beta_cols, meta_beta_col, het_p_col)
+
+if (leave) {
+  leave_pval_cols <- paste("leave", studies, method, "meta_p", sep = "_")
+  leave_beta_cols <- paste("leave", studies, method, "meta_beta", sep = "_")
+  #leave_het_p_cols <- paste("leave", studies, method, "meta_het_p", sep = "_")
+  keep_cols <- c(keep_cols, leave_pval_cols, leave_beta_cols)
+  pval_cols <- c(pval_cols, leave_pval_cols)
+  meta_pval_cols <- c(meta_pval_cols, leave_pval_cols)
+}
+
+message("Reading file ", file, " ...")
+data <- fread(file, header = T, select = keep_cols)
+
+for (pval_thresh_i in pval_thresh) {
+  
+  message("Calculating qc metrics with p-value threshold ", pval_thresh_i, " ...")
+  
+  output_prefix <- paste0(ifelse(test = is.null(opt$options$out), yes = file, no = opt$options$out), ".qc.", pval_thresh_i)
+  
+  qc_dt_i <- copy(qc_dt)
+  
+  pass <- apply(data[, ..pval_cols], 1, function(x) any(x < pval_thresh_i, na.rm = T)) & ! is.na(data[[study_pval_cols[[1]]]])
+  DATA <- data[pass]
+  
+  # Get gw signif hits
+  sig_loc_list <- list()
+  for (pval_col in pval_cols) {
```

Comment:
seems you never use other sig_log_list sumstats than the leftmost one?? In that case you only need to do the selection for that pval col.

Reply:
no biggie though as it's not that much work... just complicates the code a  bit

Reply:
Ah true. Traces of old code when I wasn't sure whether we wanted to use the other ones as well.